src/gui/qgsadvanceddigitizingcanvasitem.cpp

```diff
@@ -158,8 +150,9 @@ void QgsAdvancedDigitizingCanvasItem::paint( QPainter *painter )
     if ( mAdvancedDigitizingDockWidget->constraintAngle()->isLocked() )
     {
       painter->setPen( mLockedPen );
-      painter->drawLine( prevPointPix - canvasMaxDimension * QPointF( std::cos( a ), std::sin( a ) ),
-                         prevPointPix + canvasMaxDimension * QPointF( std::cos( a ), std::sin( a ) ) );
+      const double canvasPadding = QLineF( prevPointPix, curPointPix ).length();
+      painter->drawLine( prevPointPix + ( canvasPadding - canvasDiagonalDimension ) * QPointF( std::cos( a ), std::sin( a ) ),
```

Comment:
Shouldn't this be?

```suggestion
      painter->drawLine( prevPointPix - ( canvasPadding + canvasDiagonalDimension ) * QPointF( std::cos( a ), std::sin( a ) ),
```

(i.e. the canvasPadding is also subtracted, not added)

Reply:
I don't think. `canvasPadding` is a shift that brings the base point back into the canvas instead of `prevPointPix`. Then, the line is drawn upstream from this new point and downstream. If the user goes far away of `prevPointPix`, the line will be uselessly huge. For example, if he is at 10 000 px of `prevPointPix` with a canvas diagonal size of 1 000 px, the line will be 22 000 px long when it can be as small as 2 000 px.